packages/flutter_driver/lib/src/common/wait.dart

```diff
@@ -0,0 +1,286 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/widgets.dart';
+
+import 'message.dart';
+
+/// A Flutter Driver command that waits until a given [condition] is satisfied.
+class WaitForCondition extends Command {
+  /// Creates a command that waits for the given [condition] is met.
+  const WaitForCondition(this.condition, {Duration timeout}) : super(timeout: timeout);
+
+  /// Deserializes this command from the value generated by [serialize].
+  WaitForCondition.deserialize(Map<String, String> json)
+      : condition = _WaitConditionDecoder.deserialize(json),
+        super.deserialize(json);
+
+  /// The condition that this command shall wait for.
+  final WaitCondition condition;
+
+  @override
+  Map<String, String> serialize() => super.serialize()..addAll(condition.serialize());
+
+  @override
+  String get kind => 'waitForCondition';
+}
+
+/// A Flutter Driver command that waits until there are no more transient callbacks in the queue.
+@Deprecated('This command has been deprecated in favor of WaitForCondition. '
```

Comment:
The depreciation should also be reflected in the doc comment to ensure it shows up in the documentation.